src/types.rs

```diff
@@ -58,6 +32,52 @@ pub struct Task {
     pub finished: bool,
 }
 
+impl From<String> for Task {
```

Comment:
Is there any special reason for using owned `String`s here instead of `&str` that I am missing?
Generally you would to my knowledge only use owned `String`s when you are saving the original string inside the struct and want to move the ownership to the struct instance.

If the caller has the data as a `&str` or creates multiple task with the same `String` currently the caller would need to make a copy of the string, which ownership is passed to `Task::from` which is then dropped after parsing. This copy can be removed because you are coping the string in the parsing function anyways and do not need the ownership of it.

There is even a special trait for parsing `&str` into a struct: [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html). However this trait expects you to return a `Result` so you may use `From<&str>` instead because we don't really want to do proper error handling in this cli and panicking is fine in our usecase.

This change would also simplify the tests where you always convert a `&str` into a owned `String` in order to be able to call this function.